injection/clients/dynamicclient/dynamicclient.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamicclient
+
+import (
+	"context"
+
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+
+	"github.com/knative/pkg/injection"
+)
+
+func init() {
+	injection.RegisterClient(withDynamicClient)
+}
+
+// dynamicClientKey is used as the key for associating information
+// with a context.Context.
+type dynamicClientKey struct{}
```

Comment:
If we go with the typed struct we'd need the ability to set clients/informers etc. Thus either this key would need to be public or we provide a `Set` func

Reply:
🤔 I think we'll only need this to be public if we *don't* do that.  Still we may want it to be public for more sophisticated testing.